### ISSUE TITLE
Feature request: Typescript suppression comment aware multiline-comment-style

### ISSUE THREAD

**Issue by tkalliom** [typescript-eslint/typescript-eslint#3276]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**
When linting with the core eslint rule `multiline-comment-style`, the following is flagged:
```typescript
// Comment explaining why there is no feasible alternative to the suppression comment
// @ts-expect-error
statementWithTypeError;
```

**Expected Result**

With the hypothetical `@typescript-eslint/multiline-comment-style` rule, the above case would be an exception to requiring multiline comment syntax

**Actual Result**
With `multiline-comment-style: ["error"]`, the code above is flagged. The automatic fix produces the following code where TypeScript errors are no longer suppressed:
```typescript
/*
 * Comment explaining why there is no feasible alternative to the suppression comment
 * @ts-expect-error
 * /
statementWithTypeError;
```
**Additional Info**
N/A

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.21.0` |
| `@typescript-eslint/parser`        | `4.21.0` |
| `TypeScript`                       | `4.2.3` |
| `ESLint`                           | `7.23.0` |
| `node`                             | `10.24.0` |

**Comment by bradzacher** [comment]
Just inline your description instead so you don't need a multiline?

```ts
// @ts-expect-error my description here
```

Or use multiline comments with the last line being the directive

```ts
/*
 * my description here
 * @ts-expect-error */
```

I'll be honest with you when I say that `multiline-comment-style` is a bad rule - there's no extra clarity gained from using a block comment over multiple single line comments.
Also it's rarely used rule to begin with so I don't think it's worth the effort of creating an entire extension rule (which would likely require a fork of the rule) for this problem.

**Comment by tkalliom** [comment]
The latter suggestion contradicts the style rule, but the former is a sensible approach which didn’t occur to me. The next person to google `typescript-eslint multiline-comment-style` should find it useful as well.
